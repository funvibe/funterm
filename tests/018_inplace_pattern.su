// Test inplace bitstring pattern matching
lua.source = "Hello"

if (<<h/utf8, e/utf8, l1/utf8, l2/utf8, o/utf8>> = lua.source) {
    lua.print("Inplace pattern matched:", h, e, l1, l2, o)
}

// Test with qualified variables in pattern
lua.data = "Hi"

<<lua.first/utf8, lua.second/utf8>> = lua.data

lua.print("Qualified pattern matched:", lua.first, lua.second)

// Test with rest/binary
lua.message = "Hello World"

<<first/utf8, rest/binary>> = lua.message

lua.print("Rest pattern matched first:", first)
lua.print("Rest pattern matched rest:", rest)
