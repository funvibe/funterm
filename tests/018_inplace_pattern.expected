Inplace pattern matched: 72 101 108 108 111
Qualified pattern matched: 72 105
Rest pattern matched first: 72
Rest pattern matched rest: ello World
